Merge subjMoved & dummylaunch

sasen opened the pull request:
Trying to merge @satra's code cleanup with @akeshavan's front-end and my up-to-date-with-master backup.
Finding lots of problems. Everyone, please help! Work off of this PR if you want to be most helpful in the merging bug-hunt.

akeshavan committed:
ENH: modal during rt run with flot plot

akeshavan committed:
fix: anishas quick fix to the modal problem

akeshavan committed:
fix: small changes for dummy launch

satra committed:
sty: initial cleanup

sasen committed:
ENH/WIP: progress is now a high-water mark. if subject moved, redo the basics.

satra committed:
ref: flot code

satra committed:
enh: make button calls through a separate post - brk: not all buttons attached yet

satra committed:
fix: all buttons attached

satra committed:
fix: flot updates dynamically

satra committed:
fix: resolved conflicts

satra committed:
sty: removed extraneous comments

satra committed:
fix: js references

sasen committed:
FIX/BUG: progress is just current progress, adding timestamp check & prereqs
bug: checkbox 'checked' values don't reflect redo.

sasen committed:
FIX: checkboxes are now disabled on click, handled properly by redo logic.

sasen committed:
ENH/FIX: subjMovedLogic done
json_template.py:
-- automatically extracts prerequisites.
-- 'study_info:resume' holds savepoint during redo
buttonlib.py:
-- redo-mode (resume = a button id, not None) uses cleared timestamps
     to force redos. can skip past completed steps, then self-clears resume.

sasen committed:
FIX: setting resume for redo-within-a-redo is now handled properly.

sasen committed:
Merge branch 'master' into subjMovedLogic

sasen committed:
Merge branch 'master' into subjMovedLogic

sasen committed:
MRG/WIP/BRK: incorporating code cleanup, hooking up new layout with flots and redo logic.
Merge remote-tracking branch 'satras/dummylaunch' into mergeSubjDummy
Conflicts:
	scripts/cool_mako/realtime_app.py
	scripts/cool_mako/subreg.html

satra commented:
@sasen could you please enumerate some of the key bugs - also make sure to use chrome, not firefox. you will also need to undo those bits changed in 181030a for anything to actually work.

sasen commented:
I've been fixing things as I find them. I think some of these bugs were masked by the fact that 181030a commented out actual process launching. I've been at it for 12 hours, and I'd like to go home, but briefly:
- Undoing the changes in 181030a would've been nice. What's the point of a PR that, if accepted, wouldn't work on the production platform?
- in realtime_app.py: self.flotJavascript() was called with arguments though it is currently defined to take only self.
- I can't get the don't-click-anything modal to launch.
- The 72 404-errors per button click, given that there are git-tracked template jsons present for this exact purpose.

sasen committed:
FIX: visitDir creation, visit login takes you to the right tab.
-- login automatically takes you to the right tab.
-- if xfm and mask directories don't exist yet, login will punt you
     back to visit0.
-- createRtSession.py uses raise rather than sys.exit(1) now.

akeshavan commented:
Looks good, flot works, redo is cool, login is good! Now we need the flot modal to work. Will start that as a new PR
